app/selectors.js

```diff
@@ -278,6 +291,109 @@ export const viewedTransaction = createSelector(
   (transactions, txHash) => find({ txHash }, transactions.All)
 );
 
+export const decodedTransactions = get(["grpc", "decodedTransactions"]);
+
+const ticketNormalizer = createSelector(
+  [decodedTransactions, network],
+  (decodedTransactions, network) => {
+    return ticket => {
+      const hasSpender = ticket.spender && ticket.spender.getHash();
+      const isVote = ticket.status == "voted";
+      const ticketTx = ticket.ticket;
+      const spenderTx = hasSpender ? ticket.spender : null;
+      const hash = reverseHash(Buffer.from(ticketTx.getHash()).toString("hex"));
+      const spenderHash = hasSpender ? reverseHash(Buffer.from(spenderTx.getHash()).toString("hex")) : null;
+      const decodedTicketTx = decodedTransactions[hash] || null;
+      const decodedSpenderTx = hasSpender ? (decodedTransactions[spenderHash] || null) : null;
+
+      // effective ticket price is the output 0 for the ticket transaction
+      // (stakesubmission script class)
+      const ticketPrice = ticketTx.getCreditsList()[0].getAmount();
+
+      // ticket tx fee is the fee for the transaction where the ticket was bought
+      const ticketTxFee = ticketTx.getFee();
+
+      // missing: split tx fee
+
+      // ticket change is anything returned to the wallet on ticket purchase.
+      // (don't know if it is currently possible to have change due to split transactions)
+      const ticketChange = ticketTx.getCreditsList().slice(1).reduce((a, v) => a+v.getAmount(), 0);
```

Comment:
I **think** this is not needed as the ticket transaction should only produce the ticket commitment output, but I'm also not entirely sure, so I decided to include this here.

Reply:
Yah this will need to be smarter coming up due to the possible removal of the splittx and use of the ticket change.  But should be fine for a first pass.